sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -445,30 +446,46 @@ SHUTDOWN_FUNC(noreturn, shutdown_hang(void)) {
 #ifdef OT_PLATFORM_RV32
   while (true) {
     asm volatile(
-        "1:"
+        ".L_shutdown_hang_asm_start:"
         // Request a new scrambling key.
         "sw %[kSramRenewKey], %[kSramCtrlCtrlReg](%[kSramCtrlBase]);"
         // Request a system reset.
         "sw %[kMultiBitBool4True], %[kRstmgrResetReqReg](%[kRstmgrBase]);"
 
-        // TODO(lowRISC/opentitan#7148): restrict the ePMP such that only
-        // ROM may execute.  mundaym's suggestion: set entry 2 as a NAPOT
-        // region covering the entire address space, clear all its permission
-        // bits and set the lock bit, and then finally disable RLB to prevent
-        // any further modifications.
+        // Reconfigure ePMP such that only this inline asm is executable:
+        // - Entry 0: TOR, LRX, [.L_shutdown_hang_asm_start,
+        // .L_shutdown_hang_asm_end).
+        // - Entry 1: NAPOT, L, entire address space.
+        // - MSECCFG: MMWP set RLB unset to prevent any further modifications.
+        "la   t0, .L_shutdown_hang_asm_start;"
```

Comment:
IIRC the address needs to be shifted right by two (i.e. word aligned) before being loaded into the pmpaddrN registers.

Reply:
you're right, these addresses are bits 33-2 of a 34-bit physical address. added `"srli t0, t0, 2;"` for both addresses